src/runtime/engines/singularity/process.go

```diff
@@ -28,6 +27,80 @@ import (
 	"github.com/sylabs/singularity/src/pkg/sylog"
 )
 
+func (engine *EngineOperations) checkExec() error {
+	args := engine.EngineConfig.OciConfig.Process.Args
+	env := engine.EngineConfig.OciConfig.Process.Env
+
+	// match old behavior of searching path
+	oldpath := os.Getenv("PATH")
+	defer func() {
+		os.Setenv("PATH", oldpath)
+		engine.EngineConfig.OciConfig.Process.Args = args
+		engine.EngineConfig.OciConfig.Process.Env = env
+	}()
+
+	for _, keyval := range env {
+		if strings.HasPrefix(keyval, "PATH=") {
+			os.Setenv("PATH", keyval[5:])
+			break
+		}
+	}
+
+	if _, err := exec.LookPath(args[0]); err == nil {
```

Comment:
Would add a comment here:

`// If args[0] is an absolute path, exec.LookPath() looks for this file directly instead of within PATH`